### PR TITLE
Add new rule AvoidTooShortNames 

### PR DESCRIPTION
**Pull request by janus**
Fixes https://github.com/fsprojects/FSharpLint/issues/510

### CODE REVIEW

**Review by su8898 — COMMENTED** [review]
You need to update the FSharpLint documentation to reflect this new rule. Please check the `\docs\content\how-tos` directory and follow the pattern used to document other rules to add documentation for this new rule.

**Comment by janus** [comment]
> You need to update the FSharpLint documentation to reflect this new rule. Please check the `\docs\content\how-tos` directory and follow the pattern used to document other rules to add documentation for this new rule.

Thanks. Done.

**Comment by knocte** [comment]
> Thanks. Done.

It is not done, you forgot to include the file in your commit.

**Comment by janus** [comment]
> > Thanks. Done.
> 
> It is not done, you forgot to include the file in your commit.

Blind Bartimaeus!
Thanks. Done!

File: docs/content/how-tos/rules/FL0075.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+*Introduced in `0.21.0`*
+
+## Cause
+
+Rule to detect usage of 1 letter as variable names, function names, and parameter names.
+
+## Rationale
+
+To ensure that programmers use longer variable names than 1 letter.
```
Comment (knocte) [inline docs/content/how-tos/rules/FL0075.md:17]: but why should programmers use longer variable names? you have to explain this here, not what you wrote

File: docs/content/how-tos/rules/FL0075.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+---
+title: FL0075
+category: how-to
+hide_menu: true
+---
+
+# VariableLengthGreaterThanOne (FL0075)
+
+*Introduced in `0.21.0`*
```
Comment (knocte) [inline docs/content/how-tos/rules/FL0075.md:9]: 0.21.0 is already released, use 0.21.1

File: src/FSharpLint.Core/Text.resx [hunk 3]
```diff
@@ -174,6 +174,9 @@
   <data name="RulesNamingConventionsUnderscoreError" xml:space="preserve">
     <value>Consider changing `{0}` to remove any underscores.</value>
   </data>
+  <data name="RulesVariableLengthGreaterThanOneError" xml:space="preserve">
+    <value>Consider changing `{0}` to have minimum length of two.</value>
```
Comment (knocte) [inline src/FSharpLint.Core/Text.resx:178]: let's not mention two, just say more than one

File: tests/FSharpLint.Core.Tests/Rules/Conventions/VariableLengthGreaterThanOne.fs [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+open NUnit.Framework
+open FSharpLint.Framework.Rules
+open FSharpLint.Rules
+
+[<TestFixture>]
+type TestConventionsVariableLengthGreaterThanOne() =
+    inherit TestAstNodeRuleBase.TestAstNodeRuleBase(VariableLengthGreaterThanOne.rule)
+
+    [<Test>]
+    member this.MinimumVariableLength() =
```
Comment (knocte) [inline tests/FSharpLint.Core.Tests/Rules/Conventions/VariableLengthGreaterThanOne.fs:12]: improve this test name please

File: src/FSharpLint.Core/Rules/Conventions/VariableLengthGreaterThanOne.fs [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+open FSharpLint.Framework.Ast
+open FSharpLint.Framework.Rules
+open FSharpLint.Framework
+open FSharpLint.Framework.Suggestion
+
+let private checkIdentifierPart (identifier:Ident) (idText:string) =
+    let formatError errorName =
+        String.Format(Resources.GetString errorName, idText)
+
+    let addVariableOfLenghtOne (ident:Ident) = lazy(
```
Comment (knocte) [inline src/FSharpLint.Core/Rules/Conventions/VariableLengthGreaterThanOne.fs:14]: that's weird style, please place the lazy keyword in the next line

File: docs/content/how-tos/rules/FL0075.md (hunk 1 above)
Comment (janus) [reply docs/content/how-tos/rules/FL0075.md]: Updated.

File: docs/content/how-tos/rules/FL0075.md (hunk 2 above)
Comment (janus) [reply docs/content/how-tos/rules/FL0075.md]: Updated.

File: src/FSharpLint.Core/Text.resx (hunk 3 above)
Comment (janus) [reply src/FSharpLint.Core/Text.resx]: Updated.

File: tests/FSharpLint.Core.Tests/Rules/Conventions/VariableLengthGreaterThanOne.fs (hunk 4 above)
Comment (janus) [reply tests/FSharpLint.Core.Tests/Rules/Conventions/VariableLengthGreaterThanOne.fs]: Thanks. Done

File: src/FSharpLint.Core/Rules/Conventions/VariableLengthGreaterThanOne.fs (hunk 5 above)
Comment (janus) [reply src/FSharpLint.Core/Rules/Conventions/VariableLengthGreaterThanOne.fs]: Waa Woo. Done!

File: docs/content/how-tos/rules/FL0075.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+*Introduced in `0.21.1`*
+
+## Cause
+
+Rule to detect usage of 1 letter as variable names, function names, and parameter names.
+
+## Rationale
+
+Variable name represents the information the variable contains. It helps to improve code quality particularly during debugging. Variable length of more than one makes reading code pretty easier.
```
Comment (knocte) [inline docs/content/how-tos/rules/FL0075.md:17]: it is not that "variable length of more than one makes reading code easier", it's that variable of only one letter give poor info about what the variable contains; it's better expressed the other way around

File: docs/content/how-tos/rules/FL0075.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+## How To Fix
+
+Make sure that the length of all your variable names, function names, and parameter names are up 2 letters.
+
+## Rule Settings
+
+    {
+        "variableLengthGreaterThanOne": {
+           "enabled": true
```
Comment (knocte) [inline docs/content/how-tos/rules/FL0075.md:27]: let's make this enabled property consistent with fsharplint.json

File: tests/FSharpLint.Core.Tests/Rules/Conventions/VariableLengthGreaterThanOne.fs [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+open NUnit.Framework
+open FSharpLint.Framework.Rules
+open FSharpLint.Rules
+
+[<TestFixture>]
+type TestConventionsVariableLengthGreaterThanOne() =
+    inherit TestAstNodeRuleBase.TestAstNodeRuleBase(VariableLengthGreaterThanOne.rule)
+
+    [<Test>]
+    member this.VariableLengthMoreThanOne() =
```
Comment (knocte) [inline tests/FSharpLint.Core.Tests/Rules/Conventions/VariableLengthGreaterThanOne.fs:12]: this test still doesn't have a good name, what are you testing here?

Comment (knocte) [reply tests/FSharpLint.Core.Tests/Rules/Conventions/VariableLengthGreaterThanOne.fs:12]: (e.g. if you keep consistency with the other test names, this test name should include the word "Should")

File: src/FSharpLint.Core/Text.resx [hunk 9]
```diff
@@ -174,6 +174,9 @@
   <data name="RulesNamingConventionsUnderscoreError" xml:space="preserve">
     <value>Consider changing `{0}` to remove any underscores.</value>
   </data>
+  <data name="RulesVariableLengthGreaterThanOneError" xml:space="preserve">
+    <value>Consider changing `{0}` to more than 1 letter.</value>
```
Comment (knocte) [inline src/FSharpLint.Core/Text.resx:178]: let's use the word "single" instead of the number "1"

Comment (knocte) [reply src/FSharpLint.Core/Text.resx:178]: -> `than a single letter`

File: docs/content/how-tos/rules/FL0075.md (hunk 6 above)
Comment (janus) [reply docs/content/how-tos/rules/FL0075.md]: Updated.

File: docs/content/how-tos/rules/FL0075.md (hunk 7 above)
Comment (janus) [reply docs/content/how-tos/rules/FL0075.md]: Done.

File: tests/FSharpLint.Core.Tests/Rules/Conventions/VariableLengthGreaterThanOne.fs (hunk 8 above)
Comment (janus) [reply tests/FSharpLint.Core.Tests/Rules/Conventions/VariableLengthGreaterThanOne.fs]: Updated.

File: src/FSharpLint.Core/Text.resx (hunk 9 above)
Comment (janus) [reply src/FSharpLint.Core/Text.resx]: Updated.

**Comment by duckmatt** [comment]
Looks good! thank you

Could be worth adding the length as a config option, but more of an enhancement that could be added in the future than a must have

**Pull request merged**